internal/hooks/auth_hooks.go

```diff
@@ -93,7 +93,7 @@ const MinimumViableTokenSchema = `{
       "type": "string"
     }
   },
-  "required": ["aud", "exp", "iat", "sub", "email", "phone", "role", "aal", "session_id"]
+  "required": ["aud", "exp", "iat", "sub", "email", "phone", "role", "aal", "session_id", "is_anonymous"]
```

Comment:
```suggestion
  "required": ["aud", "exp", "iat", "sub", "email", "phone", "role", "aal",  "amr", "session_id", "is_anonymous"]
```

Reply:
The `amr` claim can be empty if a refresh is done and the [session has no AMR Claims](https://github.com/supabase/auth/blob/25d1447715cb7b562d311147c4f53d67b1012e0d/internal/models/sessions.go#L282)

I'm not sure if this is representative of a real world scenario. Will leave that to a separate PR which needs to:
1. Check if it's representative of a real world scenario
2. Update the Custom Access Token to reflect real world scenario if it is